docs/tech_specs/0005_eink_widget_endpoint.md

```diff
@@ -0,0 +1,216 @@
+```
+"flex_zone": {
+    "medium": {
+        "asset_url": "https://mbta-screens.s3.amazonaws.com/screens-prod/images/e-ink/psa/MBTA SEE SAY_Eink-Messaging.png",
+        "type": "evergreen_content"
+    },
+    "num_pages": 2,
+    "page_index": 1,
+    "type": "one_medium"
+}
+```
+
+Then the bit that should be passed in the POST request is a step deeper at the `medium` level. The request body should be:
+```
+{
+    "widget": {
+        "medium": {
+            "asset_url": "https://mbta-screens.s3.amazonaws.com/screens-prod/images/e-ink/psa/MBTA SEE SAY_Eink-Messaging.png",
+            "type": "evergreen_content"
+        }
+    }
+}
+```
+
+### Frontend rendering
+
+In both `gl_eink.tsx` and `bus_eink.tsx`, there will be a new route to match on the exact path for that screen type: e.g. `/v2/screen/gl_eink_v2/widget` or `/v2/screen/bus_eink_v2/widget`. That route will render a new type of page, `<WidgetPage />`, which takes no props. Instead, it reads the `data-widget-data` from the html app container. That value will be a string, so it will need to be parsed into json.
+
+```
+const WidgetPage = () => {
+  const widget = getDatasetValue("widgetData")
+  let widgetJson = widget ? JSON.parse(widget) : null
+  if (widgetJson) widgetJson = Object.values(widgetJson)[0]
+
+  return widgetJson ? <Widget data={widgetJson} /> : null
+};
+```
+
+And then suddenly, the widget page is rendered! Mercury will be taking screenshots of these pages and sending the image to the screen, only to update with a new screenshot if the data for that widget changes.
+
+Example 1: Footer. POST to http://localhost:4000/v2/screen/gl_eink_v2/widget with body
+```
+{
+    "widget": {
+        "footer": {
+            "mode_cost": "$2.40",
+            "mode_icon": "subway-negative-black.svg",
+            "mode_text": "Subway",
+            "text": "For real-time predictions and fare purchase locations:",
+            "type": "fare_info_footer",
+            "url": "mbta.com/stops/place-bcnwa"
+        }
+    }
+}
+```
+
+displays:
+![example eink footer][example eink footer]
+
+Example 2: Partial alert. POST to http://localhost:4000/v2/screen/gl_eink_v2/widget with body
+```
+{
+    "widget": {
+        "medium": {
+            "body": "Shuttle buses replacing Green Line E branch service",
+            "header": "Shuttle Buses",
+            "icon": "bus",
+            "route_pills": [
+                {
+                    "color": "green",
+                    "text": "Green Line E",
+                    "type": "text"
+                }
+            ],
+            "type": "alert",
+            "url": "mbta.com/alerts"
+        }
+    }
+}
+```
+
+displays:
+![example partial alert][example eink partial alert]
+
+Example 3: Takeover alert. POST to http://localhost:4000/v2/screen/gl_eink_v2/widget with body
+```
+{
+    "widget": {
+        "body": {
```

Comment:
This example is a little different because we send back data for a [region](https://github.com/mbta/screens/blob/master/docs/architecture/widget_framework.md#glossary) rather than one slot.
How would Mercury know to ask for this larger grouping of slots vs one of the individual slots? If they did ask for each individual slot in separate POSTs, would things still render correctly?

---

Ok I thought about this some more and it probably wouldn't be that hard for Mercury to do this. They could check the `type` field on each object, and if it's one they don't have code to handle, they would immediately pass that whole object including all nested values, to us for rendering.

So in this case they would see `body_takeover` and go "what even is that" and toss the data back to us.

(Leaving this comment in case you have other thoughts to add.)